ui/src/groups/useGroupJoin.ts

```diff
@@ -79,7 +82,25 @@ export default function useGroupJoin(
       try {
         await useGroupState.getState().join(flag, true);
         setStatus('success');
-        if (isTalk) {
+        if (redirectItem) {
+          if (redirectItem.type === 'chat') {
+            if (isTalk) {
+              const href = `/apps/groups/groups/${flag}/channels/${redirectItem.nest}?msg=${redirectItem.id}&joinref=true`;
+              window.open(`${window.location.origin}${href}`, '_blank');
+            } else {
+              navigate(
+                `/groups/${flag}/channels/${redirectItem.nest}?msg=${redirectItem.id}&joinref=true`
+              );
+            }
+          } else if (isTalk) {
+            const href = `/apps/groups/groups/${flag}/channels/${redirectItem.nest}/${redirectItem.type}/${redirectItem.id}&joinref=true`;
+            window.open(`${window.location.origin}${href}`, '_blank');
+          } else {
+            navigate(
+              `/groups/${flag}/channels/${redirectItem.nest}/${redirectItem.type}/${redirectItem.id}?joinref=true`
+            );
+          }
```

Comment:
this could be cleaned up a bit since these conditionals and paths are pretty similar